crates/bevy_render/src/globals.rs

```diff
@@ -0,0 +1,74 @@
+use crate::{
+    extract_resource::ExtractResource,
+    render_resource::{ShaderType, UniformBuffer},
+    renderer::{RenderDevice, RenderQueue},
+    Extract, RenderApp, RenderStage,
+};
+use bevy_app::{App, Plugin};
+use bevy_ecs::prelude::*;
+use bevy_log::info;
+use bevy_reflect::Reflect;
+use bevy_time::Time;
+
+#[derive(Default, Resource)]
+struct FrameCount(u32);
+
+pub struct GlobalsPlugin;
+
+impl Plugin for GlobalsPlugin {
+    fn build(&self, app: &mut App) {
+        if let Ok(render_app) = app.get_sub_app_mut(RenderApp) {
+            render_app
+                .init_resource::<GlobalsBuffer>()
+                .init_resource::<Time>()
+                .init_resource::<FrameCount>()
+                .add_system_to_stage(RenderStage::Extract, extract_time)
+                .add_system_to_stage(RenderStage::Prepare, prepare_globals_buffer);
+        }
+    }
+}
+
+fn extract_time(mut commands: Commands, time: Extract<Res<Time>>) {
+    commands.insert_resource(time.clone());
+}
+
+/// Contains global values useful when writing shaders.
+/// Currently only contains values related to time.
+#[derive(Default, Clone, Resource, ExtractResource, Reflect, ShaderType)]
+#[reflect(Resource)]
+pub struct GlobalsUniform {
+    /// The time since startup in seconds
+    time: f32,
+    /// The duration of the last frame in seconds
+    delta_time: f32,
+    /// Frame count since the start of the app.
+    frame_count: u32,
+}
+
+/// The buffer containing the [`GlobalsUniform`]
+#[derive(Resource, Default)]
+pub struct GlobalsBuffer {
+    pub buffer: UniformBuffer<GlobalsUniform>,
+}
+
+fn prepare_globals_buffer(
+    render_device: Res<RenderDevice>,
+    render_queue: Res<RenderQueue>,
+    mut globals_buffer: ResMut<GlobalsBuffer>,
+    time: Res<Time>,
+    mut frame_count: Local<u32>,
+) {
+    // We can't rely on the FrameTimeDiagnosticsPlugin to be present since it is not part of DefaultPlugins
+    *frame_count = frame_count.wrapping_add(1);
+
+    let buffer = globals_buffer.buffer.get_mut();
+    buffer.time = time.seconds_since_startup() as f32;
+    buffer.delta_time = time.delta_seconds();
+    buffer.frame_count = *frame_count;
+
+    info!("{frame_count:?}");
```

Comment:
If this just prints the frame number value and nothing else, it probably needs elaborating. However, I'm not sure we should have such a verbose message in info?

Reply:
😅 that wasn't supposed to be committed, I was just debugging. I'll remove that